src/main/java/com/example/bookstore/service/user/UserServiceImpl.java

```diff
@@ -0,0 +1,33 @@
+package com.example.bookstore.service.user;
+
+import com.example.bookstore.dto.user.UserRegistrationRequestDto;
+import com.example.bookstore.dto.user.UserResponseDto;
+import com.example.bookstore.exception.RegistrationException;
+import com.example.bookstore.mapper.UserMapper;
+import com.example.bookstore.model.User;
+import com.example.bookstore.repository.user.UserRepository;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+@RequiredArgsConstructor
+@Service
+public class UserServiceImpl implements UserService {
+    private final UserRepository userRepository;
+    private final UserMapper userMapper;
+
+    @Override
+    public UserResponseDto register(UserRegistrationRequestDto requestDto)
+            throws RegistrationException {
+        if (userRepository.findByEmail(requestDto.getEmail()).isPresent()) {
```

Comment:
```suggestion
        if (userRepository.existsByEmail(requestDto.getEmail())) {
```
It's better approach